### ISSUE TITLE
GPG key error

### ISSUE THREAD

**Issue by xhdix** [TheTorProject/ooni-web#258]
To install OONI in Debian, there is a problem getting GPG key.
https://github.com/TheTorProject/ooni-web/blob/master/content/install/ooniprobe.md

```
root@dbn:/home/user# gpg --keyserver keys.gnupg.net --recv A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89
gpg: keyserver receive failed: No keyserver available
root@dbn:/home/user# gpg --keyserver keys.gnupg.net --recv A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89
gpg: keyserver receive failed: No keyserver available
root@dbn:/home/user# apt-key adv --keyserver keys.gnupg.net --recv-key A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89     
Executing: /tmp/apt-key-gpghome.6FwcbNgQd8/gpg.1.sh --keyserver keys.gnupg.net --recv-key A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89
gpg: keyserver receive failed: No keyserver available
root@dbn:/home/user# apt-key adv --keyserver hkp://keys.gnupg.net --recv-key A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89
Executing: /tmp/apt-key-gpghome.KTXyWIAuiM/gpg.1.sh --keyserver hkp://keys.gnupg.net --recv-key A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89
gpg: keyserver receive failed: No keyserver available
```

Previously, the solution to this problem was to use the following commands:

```
curl https://deb.torproject.org/torproject.org/A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89.asc | gpg --import
​
gpg --export A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89 | apt-key add - 
```

Result:

```
root@dbn:/home/user# curl https://deb.torproject.org/torproject.org/A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89.asc | gpg --import
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 19665  100 19665    0     0   4702      0  0:00:04  0:00:04 --:--:--  5011
gpg: key EE8CBC9E886DDD89: 36 signatures not checked due to missing keys
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key EE8CBC9E886DDD89: public key "deb.torproject.org archive signing key" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: no ultimately trusted keys found
root@dbn:/home/user# gpg --export A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89 | apt-key add -
OK
```

I think the reference to the solution on that page is good.
And the question is, why is it so? :)

**Comment by bassosimone** [comment]
Does the problem persists when using other keyservers? Like hkps://keyserver.ubuntu.com?

**Comment by xhdix** [comment]
Yes..

```
root@dbn:/home/user# apt-key adv --keyserver hkps://keyserver.ubuntu.com --recv-key A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89
Executing: /tmp/apt-key-gpghome.UYPMTKzBvU/gpg.1.sh --keyserver hkps://keyserver.ubuntu.com --recv-key A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89
gpg: packet(13) too large
gpg: read_block: read error: Invalid packet
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
```

I have already tried various solutions on different operating systems, but there will always be a problem that makes it impossible to get the Tor key.

**Comment by bassosimone** [comment]
That goes way beyond my PGP depth. I do not know what to say because I do not understand the issue. The only clue is that a packet is “too large” (I assume there is some tag-length-value somewhere but that’s really an assumption). This is driving me to curiosity now, however: is there anyone else reporting this kind of error for gpg?

**Comment by xhdix** [comment]
Yes..
In the last year, I've repeatedly touched this problem and each time the conditions get worse.
However, you can find lots of results on Google.
https://www.google.com/search?q=tor+GPG+error&tbs=qdr:y  (Limited to past year)

But the solution that has always solved the problem was something I said and someone mentioned here too:
https://github.com/Stadicus/guides/issues/343#issuecomment-466739938

Of course, it's not a logical solution.

**Comment by xhdix** [comment]
This method has also changed on the Tor site:
https://2019.www.torproject.org/docs/debian.html.en

**Comment by p1g30n** [comment]
@bassosimone I'm getting the same in errors from the ubuntu keyserver and ```gpg: keyserver receive failed: No data```from gnupg.net in Debian 9.8

**Comment by bassosimone** [comment]
Thanks guys :). Yeah, I think we should probably change the page to be consistent with what Tor recommends to do as mentioned here https://github.com/TheTorProject/ooni-web/issues/258#issuecomment-478144894

**Comment by xhdix** [comment]
fixed with #259
